File: simple/csharp/introduce-null-object_after2.cs
class NullConnection extends Connection 
{
  boolean isNull() 
  {
    return true;
  }

  Connection getStandartConnection() 
  {
    return new HighSpeedConnection();
  }
  // Some other NULL functionality.
}

// Replace null values with Null-object.
connection = (session.connection != null) ? 
  session.connection : new NullConnection();

File: simple/php/introduce-null-object_after2.php
class NullConnection extends Connection {
  function isNull() {
    return true;
  }

  function getStandartConnection() {
    return new HighSpeedConnection();
  }
  // Some other NULL functionality.
}

// Replace null values with Null-object.
$connection = ($session.connection != null) ? 
  $session->connection : 
  new NullConnection();
